internal/rebooter/reconcile.go

```diff
@@ -134,20 +138,40 @@ func (r *RebooterReconciler) checkIfNodeNeedsDrain(ctx context.Context, node *co
 
 // checkIfNodeNeedsReboot checks if the node with the given name needs to be rebooted.
 func (r *RebooterReconciler) checkIfNodeNeedsReboot(ctx context.Context, node *corev1.Node) bool {
-	logger := log.FromContext(ctx).WithName("CheckIfNodeNeedsReboot").WithValues("nodeName", node.Name)
+	logger := log.FromContext(ctx).WithName("CheckIfNodeNeedsReboot").WithValues("nodeName", node.Name).V(1)
 
 	if r.CheckNodeCondition(ctx, node, consts.SlurmNodeReboot, corev1.ConditionTrue) {
-		logger.V(1).Info("Need to reboot node")
-		return true
+		logger.Info("Need to reboot node")
+		if nodeCondSlurmReboot := r.GetNodeConditions(ctx, node, consts.SlurmNodeReboot); nodeCondSlurmReboot != nil {
+			if r.IsUptimeGreaterThanLastTransition(ctx, nodeCondSlurmReboot.LastTransitionTime) {
+				logger.Info("Uptime is greater than last transition time")
+				return true
+			}
+		}
```

Comment:
NIT: I think there's even no need for `if r.CheckNodeCondition(ctx, node, consts.SlurmNodeReboot, corev1.ConditionTrue) {` line, just get node condition and check it, to not iterate over conditions twice (but it's not crucial anyway)
```suggestion
		nodeCondSlurmReboot := r.GetNodeConditions(ctx, node, consts.SlurmNodeReboot)
		if nodeCondSlurmReboot == nil {
			return false;
		}
		logger.WithValues("nodeCondition", nodeCondSlurmReboot).Info("Checking if node needs reboot)
		return nodeCondition.Status == corev1.ConditionTrue && 
		  r.IsUptimeGreaterThanLastTransition(ctx, nodeCondSlurmReboot.LastTransitionTime)
```